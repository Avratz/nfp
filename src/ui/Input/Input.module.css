.input {
  background-color: transparent;
  border-radius: 0.75rem;
  font-family: var(--font-poppins);
  font-weight: var(--font-medium);
  font-size: var(--text-sm);
  color: var(--neutral-8);
  padding: 0.75rem 1rem;
  width: 100%;
}

.input::placeholder {
  color: var(--neutral-8);
  opacity: 1;
}

.inputLabel {
  display: block;
}
.inputWrapper {
  position: relative;
}

.base {
  border: 2px solid var(--neutral-3);
}

.blank {
  border: none;
  border-radius: 0;
}

.blank:focus {
  outline: none;
}

.icon {
  position: absolute;
  top: 0;
  color: var(--neutral-4);
}

.iconLeft {
  left: 1rem;
}

.iconRight {
  right: 1rem;
}

.inputLeft {
  padding-left: 3.5rem;
}

.inputRight {
  padding-right: 3.5rem;
}
