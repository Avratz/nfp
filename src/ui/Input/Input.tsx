'use client'
import { cva, cx } from 'class-variance-authority'
import { useId, InputHTMLAttributes } from 'react'
import { Icon, IconsName } from '@/ui/Icon/Icon'

import styles from './Input.module.css'

const inputStyles = cva(styles.input, {
  variants: {
    intent: {
      base: styles.base,
      blank: styles.blank,
    },
  },
  defaultVariants: {
    intent: 'base',
  },
})

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  intent?: 'base' | 'blank'
  label?: string
  icon?: {
    slot: 'left' | 'right'
    name: IconsName
    class?: string
  }
}

export function Input({
  intent,
  label,
  className = '',
  icon,
  id,
  ...props
}: InputProps) {
  const _id = useId()
  return (
    <>
      <label htmlFor={id || _id} className={styles.inputLabel}>
        {label ? label : null}
        <div className={styles.inputWrapper}>
          {icon?.slot === 'left' && icon?.name !== undefined ? (
            <Icon
              name={icon?.name}
              className={cx(styles.icon, styles.iconLeft, icon.class)}
            />
          ) : null}
          <input
            className={inputStyles({
              intent,
              className: cx(
                icon?.slot === 'left'
                  ? styles.inputLeft
                  : icon?.slot === 'right'
                  ? styles.inputRight
                  : null,
                className,
              ),
            })}
            id={id || _id}
            type="text"
            {...props}
          />
          {icon?.slot === 'right' && icon?.name !== undefined ? (
            <Icon
              name={icon?.name}
              className={cx(styles.icon, styles.iconRight, icon.class)}
            />
          ) : null}
        </div>
      </label>
    </>
  )
}
