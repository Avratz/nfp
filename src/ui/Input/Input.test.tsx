import { render } from '@testing-library/react'
import { Input } from './Input'

describe('Input', () => {
  it('renders default', () => {
    const { container } = render(<Input />)
    expect(container).toMatchSnapshot()
  })

  it('renders with label', () => {
    const { container } = render(<Input label="Label" />)
    expect(container).toMatchSnapshot()
  })

  it('renders with placeholder', () => {
    const { container } = render(<Input placeholder="Placeholder" />)
    expect(container).toMatchSnapshot()
  })

  it('renders with intent blank', () => {
    const { container } = render(<Input intent="blank" />)
    expect(container).toMatchSnapshot()
  })
})
