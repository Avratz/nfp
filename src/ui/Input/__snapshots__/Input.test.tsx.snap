// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders default 1`] = `
<div>
  <label
    class="inputLabel"
    for=":r0:"
  >
    <div
      class="inputWrapper"
    >
      <input
        class="input base"
        id=":r0:"
        type="text"
      />
    </div>
  </label>
</div>
`;

exports[`Input renders with intent blank 1`] = `
<div>
  <label
    class="inputLabel"
    for=":r3:"
  >
    <div
      class="inputWrapper"
    >
      <input
        class="input blank"
        id=":r3:"
        type="text"
      />
    </div>
  </label>
</div>
`;

exports[`Input renders with label 1`] = `
<div>
  <label
    class="inputLabel"
    for=":r1:"
  >
    Label
    <div
      class="inputWrapper"
    >
      <input
        class="input base"
        id=":r1:"
        type="text"
      />
    </div>
  </label>
</div>
`;

exports[`Input renders with placeholder 1`] = `
<div>
  <label
    class="inputLabel"
    for=":r2:"
  >
    <div
      class="inputWrapper"
    >
      <input
        class="input base"
        id=":r2:"
        placeholder="Placeholder"
        type="text"
      />
    </div>
  </label>
</div>
`;
